### PR TITLE
Destination Redshift: Adapting to Kotlin CDK 

### PR DESCRIPTION
**Pull request by gisripa**
### TL;DR

Use kotlin converted CDK in destination-redshift

### CODE REVIEW

**Comment by gisripa** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36589?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#36589** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36589?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#36588** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36588?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftDestinationAcceptanceTest.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ protected JsonNode getFailCheckConfig() {
   }
 
   @Override
-  protected TestDataComparator getTestDataComparator() {
+  public TestDataComparator getTestDataComparator() {
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftDestinationAcceptanceTest.java:75]: this is probably unnecessary, earlier CDK removed the protected visibility, now fixed.

File: airbyte-integrations/connectors/destination-redshift/build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ plugins {
 airbyteJavaConnector {
     cdkVersionRequired = '0.25.0'
     features = ['db-destinations', 's3-destinations', 'typing-deduping']
-    useLocalCdk = false
+    useLocalCdk = true
```
Comment (jdpgrailsdev) [inline airbyte-integrations/connectors/destination-redshift/build.gradle:9]: We don't want to commit this change, correct?

Comment (gisripa) [reply airbyte-integrations/connectors/destination-redshift/build.gradle:9]: Nope

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java [hunk 3]
```diff
@@ -95,7 +95,8 @@ private boolean isEphemeralKeysAndPurgingStagingData(final JsonNode config, fina
   public AirbyteConnectionStatus check(final JsonNode config) {
     final S3DestinationConfig s3Config = getS3DestinationConfig(findS3Options(config));
     final EncryptionConfig encryptionConfig =
-        config.has(UPLOADING_METHOD) ? EncryptionConfig.fromJson(config.get(UPLOADING_METHOD).get(JdbcUtils.ENCRYPTION_KEY)) : new NoEncryption();
+        config.has(UPLOADING_METHOD) ? EncryptionConfig.Companion.fromJson(config.get(UPLOADING_METHOD).get(JdbcUtils.ENCRYPTION_KEY))
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java:98]: make those @Jvmstatic in the parent PR

**Review by stephane-airbyte — CHANGES_REQUESTED** [review]
don't reference the companion object. Instead, mark the CDK functions @JvmStatic

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java (hunk 3 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java:98]: Its already converted to companion object function.  Why mark it `@JvmStatic` ?

Comment (gisripa) [reply airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java]: Fixed

**Review by stephane-airbyte — APPROVED** [review]

**Comment by gisripa** [comment]
Runs with pre-release images:
https://cloud.airbyte.com/workspaces/2c817306-57f8-4b59-bf34-d8784680946c/connections/1171c486-e2a6-4a9e-ac91-5972ab4e081d/job-history#9983389::0
https://cloud.airbyte.com/workspaces/2c817306-57f8-4b59-bf34-d8784680946c/connections/3c87bf01-ae2f-4ab0-a194-1a238003437c/job-history#9983506::0

Verified record counts

**Pull request merged**